app/AppKernel.php

```diff
@@ -222,12 +223,53 @@ public function registerContainerConfiguration(LoaderInterface $loader)
             $container->setParameter('container.autowiring.strict_mode', true);
             $container->setParameter('container.dumper.inline_class_loader', false);
             $container->setParameter('prestashop.module_dir', _PS_MODULE_DIR_);
-            /** @deprecated kernel.active_modules is deprecated. Use prestashop.active_modules instead. */
+            /* @deprecated kernel.active_modules is deprecated. Use prestashop.active_modules instead. */
             $container->setParameter('kernel.active_modules', $activeModules);
             $container->setParameter('prestashop.active_modules', $activeModules);
             $container->setParameter('prestashop.installed_modules', $installedModules);
             $container->addObjectResource($this);
             $container->setParameter('modules_translation_paths', $moduleTranslationsPaths);
+
+            // Define parameter for admin folder path
+            if (defined('PS_ADMIN_DIR') && is_dir(PS_ADMIN_DIR)) {
+                $adminDir = PS_ADMIN_DIR;
+            } elseif (defined('_PS_ADMIN_DIR_') && is_dir(_PS_ADMIN_DIR_)) {
+                $adminDir = _PS_ADMIN_DIR_;
+            } else {
+                // Look for potential admin folders, condition to meet:
+                //  - first level folders in the project folder
+                //  - contains a PHP file that define the const PS_ADMIN_DIR or _PS_ADMIN_DIR_
+                //  - the first folder found is used (alphabetical order, but files named index.php have the highest priority)
+                $finder = new Symfony\Component\Finder\Finder();
+                $finder->files()
+                    ->name('*.php')
+                    ->contains('/define\([\'\"](_)?PS_ADMIN_DIR(_)?[\'\"]/')
+                    ->depth('== 1')
+                    ->sort(function (SplFileInfo $a, SplFileInfo $b): int {
+                        // Prioritize files named index.php
+                        if ($a->getFilename() === 'index.php') {
+                            return -1;
+                        }
+
+                        return strcmp($a->getRealPath(), $b->getRealPath());
+                    })
+                    ->in($this->getProjectDir())
+                ;
+                foreach ($finder as $adminIndexFile) {
+                    $adminDir = $adminIndexFile->getPath();
+                    // Container freshness depends on this file existence
+                    $container->addResource(new FileExistenceResource($adminIndexFile->getRealPath()));
```

Comment:
nice, I didn't know about this `FileExistenceResource` feature

Reply:
Yep me neither, just found out about it!